src/url.zig

```diff
@@ -85,25 +85,27 @@ pub const URL = struct {
         base: []const u8,
         comptime opts: StitchOpts,
     ) !StitchReturn(opts) {
-        if (base.len == 0 or isCompleteHTTPUrl(path)) {
-            return simpleStitch(allocator, path, opts);
+        const trimmed_path = std.mem.trim(u8, path, &.{ '\n', '\r' });
```

Comment:
could save a bit of changes by renaming the `path` parameter and use a `const path = ` here instead.